### PR TITLE
ROX-22720: Add compliance operator installed flag

### PR DESCRIPTION
**Pull request by SimonBaeumer**
## Description

This PR adds a flag to mark whether the compliance operator is installed or not. This can be used by the UI to mark CO integrations as disabled/enabled.

Also includes a minor refactoring to reduce some if-branches to get the compliance operator deployment.

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [] ~~Evaluated and added CHANGELOG entry if required~~
- [ ] ~~Determined and documented upgrade steps~~
- [ ] ~~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

 - Install StackRox (central+ sensor, via `./deploy/...` scripts)
- Query the compliance integrations API, `operatorInstalled: false` should be set. 
```
$ roxcurl /v2/compliance/integrations
{"integrations":[{"id":"9e97b164-3177-4afc-b1bf-13b9675a1db5","version":"","clusterId":"49abcaaa-4260-417e-8568-b78221a4b5cd","clusterName":"remote","namespace":"","statusErrors":["deployment compliance-operator not found in any namespace"],"operatorInstalled":false}]}
```
- Clone https://github.com/stackrox/fake-compliance-operator
- Run `./apply.sh init`
- Run `roxcurl /v2/compliance/integrations` again, should be `operatorInstalled: true` now in the response

### CODE REVIEW

File: proto/storage/compliance_integration.proto [hunk 1]
```diff
@@ -5,12 +5,13 @@ package storage;
...
-// Next Tag: 6
+// Next Tag: 7
 message ComplianceIntegration {
   string id = 1; // @gotags: sql:"pk,type(uuid)"
   string version = 2; // @gotags: search:"Compliance Operator Version,hidden,store"
   string cluster_id = 3; // @gotags: search:"Cluster ID,hidden,store" sql:"fk(Cluster:id),no-fk-constraint,type(uuid),index=category:unique;name:compliance_unique_indicator"
   string compliance_namespace = 4;
   // Collection of errors that occurred while trying to obtain compliance health info.
   repeated string status_errors = 5;
+  bool operator_installed = 6;
```
Comment (dashrews78) [inline proto/storage/compliance_integration.proto:16]: Thinking ahead we could make this a searchable field so anyone calling the `ListComplianceIntegrations` could narrow their responses

Comment (dashrews78) [reply proto/storage/compliance_integration.proto:16]:
```
bool operator_installed = 6; // @gotags: search:"Operator Installed,hidden"
```

`store` is not longer required there.  I would also recommend making the tag `Compliance Operator Installed`

If you wanted to reference the tag in code and/or a unit test you would also need to make a corresponding update to https://github.com/stackrox/stackrox/blob/bac6c6ab0c48d6f9bbac1ffbdfac194137c55d94/pkg/search/options.go#L209

Comment (SimonBaeumer) [reply proto/storage/compliance_integration.proto]: Added it :+1:

**Review by dashrews78 — APPROVED** [review]

**Pull request merged**